include/xgboost/host_device_vector.h

```diff
@@ -135,7 +135,9 @@ class HostDeviceVector {
 
   void SetDevice(DeviceOrd device) const;
 
-  void Resize(size_t new_size, T v = T());
+  void Resize(std::size_t new_size);
+  /** @brief Resize and initialize the data if the new size is larger than the old size. */
+  void Resize(size_t new_size, T v);
```

Comment:
Should this also be `std::size_t` to be consistent?

Reply:
Good catch, will fix.